realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -0,0 +1,340 @@
+ * limitations under the License.
+ */
+
+package io.realm.processor;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.processing.Messager;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.PackageElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Types;
+
+import io.realm.annotations.Ignore;
+import io.realm.annotations.Index;
+import io.realm.annotations.PrimaryKey;
+
+/**
+ * Utility class for holding metadata for RealmProxy classes.
+ */
+public class ClassMetaData {
+
+    private final TypeElement classType;
+    private String className;
+    private String packageName;
+    private boolean hasDefaultConstructor = false;
+    private VariableElement primaryKey = null;
+    private List<VariableElement> fields = new ArrayList<VariableElement>();
+    private List<VariableElement> indexedFields = new ArrayList<VariableElement>();
+    private Set<VariableElement> ignoredFields = new HashSet<VariableElement>();
+    private Set<String> expectedGetters = new HashSet<String>();
+    private Set<String> expectedSetters = new HashSet<String>();
+    private Set<ExecutableElement> methods = new HashSet<ExecutableElement>();
+    private Map<String, String> getters = new HashMap<String, String>();
+    private Map<String, String> setters = new HashMap<String, String>();
+    private final List<TypeMirror> validPrimaryKeyTypes;
+    private final Types typeUtils;
+
+    public ClassMetaData(ProcessingEnvironment env, TypeElement clazz) {
+        this.classType = clazz;
+        this.className = clazz.getSimpleName().toString();
+        typeUtils = env.getTypeUtils();
+        TypeMirror stringType = env.getElementUtils().getTypeElement("java.lang.String").asType();
+        validPrimaryKeyTypes = Arrays.asList(
+                stringType,
+                typeUtils.getPrimitiveType(TypeKind.SHORT),
+                typeUtils.getPrimitiveType(TypeKind.INT),
+                typeUtils.getPrimitiveType(TypeKind.LONG)
+        );
+    }
+
+    /**
+     * Build the meta data structures for this class. Any errors or messages will be
+     * posted on the provided Messager.
+     *
+     * @return True if processing should continue, false otherwise. Note that true might be returned for smaller errors,
+     * which will enable the annotation processor to output error messages for multiple classes before exiting.
+     */
+    public boolean generateMetaData(Messager messager) {
```

Comment:
This is still a HUGE method. Couldn't his be simplified for better readability? Better comments above each code section could at least help.